assets/js/ubi.js

```diff
@@ -164,23 +147,48 @@ export function getPageId(){
   return location.pathname;
 }
 
+function getTrail(){
+  let trail = sessionStorage.getItem('trail');
+  if(trail == null)
+    return '';
+  return trail;
+}
+
+function setTrail(){
+  let trail = getTrail();
+  if(trail && trail.length > 0){
+    if(!trail.startsWith(window.location.pathname)){
+      trail += ` › ${window.location.pathname}`;
+    }
+  } else {
+    trail = window.location.pathname;
+  }
+  sessionStorage.setItem('trail', trail);
+
+  return trail;
+}
+
```

Comment:
We would be better off with a reliable structure. Also, `startsWith` logic doesn't sound right. How about:
```suggestion
function getTrail(){
  try {
    const trail = JSON.parse(sessionStorage.getItem('trail'));
    if (Array.isArray(trail)) return trail;
  } catch (ex) { /* Do nothing */ }

  return [];
}

function setTrail(){
  const trail = getTrail();
  // No need to add the current pathname if it is already the last element in trail
  if (trail.length && trail[trail.length - 1] === window.location.pathname)
    return trail;

  sessionStorage.setItem('trail', trail.concat(window.location.pathname));

  return trail;
}

```